packages/request-client.js/src/api/payment-network/eth/info-retriever.ts

```diff
@@ -3,58 +3,55 @@ import * as Types from '../../../types';
 
 /**
  * Gets a list of transfer events for an address and payment reference
- *
- * @param address Address to check
- * @param eventName Indicate if it is an address for payment or refund
- * @param network The id of network we want to check
- * @param paymentReference The reference to identify the payment
- * @param etherscanApiToken The etherscan API token
  */
-async function getTransferEvents(
-  toAddress: string,
-  eventName: Types.EVENTS_NAMES,
-  network: string,
-  paymentReference: string,
-  etherscanApiKey?: string,
-): Promise<{
-  events: Array<{
-    name: Types.EVENTS_NAMES;
-    parameters: {
-      amount: string;
-      block?: number;
-      confirmation?: number;
-      timestamp?: number;
-      txHash?: string;
-    };
-  }>;
-}> {
-  if (network === 'private') {
-    throw new Error(
-      'ETH input data info-retriever works with etherscan and cannot work on a local network',
-    );
-  }
-  const provider = new ethers.providers.EtherscanProvider(network, etherscanApiKey);
-  const history = await provider.getHistory(toAddress);
+export default class ETHInfoRetriever
```

Comment:
I'm not a big fan of this. I like how the `info-retriever` are functional. 
I don't see the need for a class here. We can discuss more about it.

Reply:
discussed: coherence with code base + preparation of refactoring of payment detection